src/game/Warden/Warden.h

```diff
@@ -95,6 +95,43 @@ struct WardenHashRequest
     uint8 Seed[16];
 };
 
+namespace WardenState
+{
+    enum Value
+    {
+        STATE_INITIAL,
+        STATE_REQUESTED_MODULE,
+        STATE_SENDED_MODULE,
+        STATE_REQUESTED_HASH,
+        STATE_INITIALIZE_MODULE,
+        STATE_REQUESTED_DATA,
+        STATE_RESTING
+    };
+
+    inline char* to_string(WardenState::Value value)
+    {
+        switch (value)
+        {
+            case WardenState::STATE_INITIAL:
+                return "STATE_INITIAL";
+            case WardenState::STATE_REQUESTED_MODULE:
+                return "STATE_REQUESTED_MODULE";
+            case WardenState::STATE_SENDED_MODULE:
+                return "STATE_SENDED_MODULE";
+            case WardenState::STATE_REQUESTED_HASH:
+                return "STATE_REQUESTED_HASH";
+            case WardenState::STATE_INITIALIZE_MODULE:
+                return "STATE_INITIALIZE_MODULE";
+            case WardenState::STATE_REQUESTED_DATA:
+                return "STATE_SENDED_DATA";
+            case WardenState::STATE_RESTING:
+                return "STATE_RESTING";
+            default:
+                return "UNDEFINED STATE";
```

Comment:
I didnt say the code was wrong. Just that compilers can give warnings if cases are missing.
Here is an example:
No break: https://github.com/Rochet2/smallfolk_cpp/commit/31dd3a133b6b14054640c9912cc7e16db7f4cefa
Added new enum value without changing the switch: https://github.com/Rochet2/smallfolk_cpp/commit/802b86c8c3499db098c5a6a2fc697913812a64c3
And travis compile warnings (Im using the -Wall flag): https://travis-ci.org/Rochet2/smallfolk_cpp/jobs/202598262#L305-L308